src/patterns/ec2-app/framework.ts

```diff
@@ -4,11 +4,31 @@ The plan is to sprinkle some framework specific tooling into them.
 For example, a Play app should come with the infrastructure for https://github.com/guardian/play-secret-rotation.
  */
 
+import { AccessScope } from "../../constants";
 import type { GuStack } from "../../constructs/core";
+import type { GuDomainName } from "../../types";
 import type { GuEc2AppProps } from "./base";
 import { GuEc2App } from "./base";
 
-type GuEc2FrameworkAppProps = Omit<GuEc2AppProps, "applicationPort">;
+type GuEc2FrameworkAppProps = Omit<GuEc2AppProps, "applicationPort"> & { certificateProps: GuDomainName };
+
+type GuEc2WorkerProps = Omit<GuEc2AppProps, "applicationPort" | "access">;
+
+/**
+ * Creates an instance of [[`GuEc2App`]], with an application port of 9000 and no access to the load balancer.
+ * This is useful for applications that only have an LB for health check purposes
+ */
+export class GuPlayWorkerApp extends GuEc2App {
```

Comment:
I'd be tempted to not restrict this to Play, and name it `GuEc2WorkerApp` instead. However, we can make this change later once we understand if there are non Play use-cases.

Reply:
I share your temptation, but also think we should only change it if there turn out to be non-Play use-cases